uni/lib/view/library/widgets/reservation_row.dart

```diff
@@ -0,0 +1,169 @@
+import 'package:flutter/material.dart';
+import 'package:intl/date_symbol_data_local.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/main.dart';
+import 'package:uni/model/entities/library_reservation.dart';
+import 'package:uni/model/providers/lazy/library_reservations_provider.dart';
+import 'package:uni/model/providers/startup/session_provider.dart';
+import 'package:uni/view/common_widgets/toast_message.dart';
+import 'package:uni/view/locale_notifier.dart';
+
+class ReservationRow extends StatelessWidget {
+  ReservationRow(this.reservation, {super.key}) {
+    hoursStart = DateFormat('HH:mm').format(reservation.startDate);
+    hoursEnd = DateFormat('HH:mm')
+        .format(reservation.startDate.add(reservation.duration));
+    day = DateFormat('dd').format(reservation.startDate);
+    initializeDateFormatting();
+    month = DateFormat('MMMM', 'pt').format(reservation.startDate);
+  }
+  final LibraryReservation reservation;
+  late final String hoursStart;
+  late final String hoursEnd;
+  late final String weekDay;
+  late final String day;
+  late final String month;
```

Comment:
Prefer to use DateTime and display as you wish below. Also, I don't think this needs to be final